source/extensions/transport_sockets/tls/utility.cc

```diff
@@ -214,6 +215,50 @@ absl::optional<std::string> Utility::getLastCryptoError() {
   return absl::nullopt;
 }
 
+absl::string_view Utility::getErrorDescription(int err) {
```

Comment:
Drive by: It seems like this is a function that could be provided by BoringSSL? cc @PiotrSikora @davidben @ggreenway

Reply:
It is. `SSL_error_description` :-)

Reply:
OK, thanks. @Shikugawa can you please do a follow up PR to swap this out?

Reply:
@davidben @mattklein123 I found a problem that SSL_error_description can't be utilized when BORINGSSL_FIPS. I didn't survey about this issue but considered this approach would be no problem and more performant with string_view.

Reply:
That function isn't gated on FIPS mode in BoringSSL. I'm guessing Envoy pins to a much much older version in some build configurations? I'm not sure what you all's policy on that is. @PiotrSikora? `SSL_error_description` was added way back in August of 2019.

If callers want to support a range of versions, our usual recommendation is to add preprocessor checks on `BORINGSSL_API_VERSION`, so that you can still call into newer functions when they're available, and then to retire the old paths when no longer needed. In particular, `SSL_error_description` will pick up new values when we add them. So that may be an option.

Reply:
@davidben BoringSSL-FIPS builds are pinned to the latest FIPS-validated version for Linux x86-64, which per [crypto/fipsmodule/FIPS.md](https://github.com/google/boringssl/blob/master/crypto/fipsmodule/FIPS.md) is `fips-20190808`, tagged ~2 weeks before `SSL_error_description` was added.

Reply:
OK @Shikugawa can you possibly add a TODO so thast when we bump the FIPS version we can remove this function?

Reply:
Sure